scipy/special/_legacy.pxd

```diff
@@ -35,6 +36,13 @@ cdef inline void _legacy_cast_check(char *func_name, double x, double y) nogil:
                                "floating point number truncated to an integer",
                                1)
 
+cdef inline void _legacy_deprecation(char *func_name, double x, double y) nogil:
+    if <int>y != y:
+        with gil:
+            PyErr_WarnEx_noerr(DeprecationWarning,
+                               "non-integer arg is deprecated, removed in SciPy 1.6.x",
```

Comment:
You're going to want to warn whenever the `double`, `double`, `double` signature is hit, since in the future just passing a double for `n` will fail, not just a double that doesn't happen to be an integer.

Reply:
Not sure I follow how you'd like line 40 to read, are you suggesting something like: 
`if isinstance(x, double) and isinstance(y, double) and  isinstance(z, double):`
not sure if the syntax is fully correct, I'm less familiar w/Cython than I am with C or Python.

Reply:
Right, so the ufunc itself will handle dispatching to this signature, which means that by the time we hit this code we know that we are already in the case where `n` is a float, and that means we want to warn. So we don't need the conditional.

Reply:
Ok, sounds like were on the same page then, thanks.